### ISSUE TITLE
Discuss frameworks

### ISSUE THREAD

**Issue by ghost** [LoungeCPP/loungesome-cpp#1]
Boost aside, there's a few frameworks that might be worth the list, but we should probably discuss it with both people who like it and people who dislike it.

I propose discussing
- Cinder
- JUCE
- Qt (!)

**Comment by nabijaczleweli** [comment]
:-1: for Qt from me

**Comment by Morwenn** [comment]
I don't like the full OOP philosophy of Qt but I have yet to find something more complete and easy to use (ok, if we don't take `moc` and the additional build steps into account) when it comes to creating GUIs. Also, you can find plenty of help and information on the internet, which is valuable by itself.

**Comment by TheCatPlusPlus** [comment]
GTK with gtkmm has always been better at being C++ than Qt.

**Comment by ghost** [comment]
I'm not a big fan of how Qt has the tendency to take over all of a program's design and structure. also moc, ugh. :-1: for me too.

**Comment by ghost** [comment]
I should add that Qt is almost a different language than C++, compatible, but still different. A bit like Objective-C++.

**Comment by Morwenn** [comment]
Haha, it's true that when I want to use Qt, I simply switch to Python so that I don't have to deal with `QString`, `QVariant`, `moc` and some other annoying things. The signals thing was interesting but there are tons of libraries implementing signals without preprocessing tricks now.

Basically it's a usable tool, but indeed not « awesome C++ ».

**Comment by milleniumbug** [comment]
Qt is so annoying I preferred writing my GUI programs in _freaking FLTK_ (which uses `const char*` and callbacks with `void*`). OTOH I've yet to try Nana, which uses C++11 and advertises itself as modern.

As for Cinder, @andyprowl might have an opinion about it, he did a snake game in it.

**Comment by nabijaczleweli** [comment]
ISTR Cinder being trivial to pick up and use (small-scale experiment on a C++ convention).

**Comment by kbok** [comment]
Someone volunteer to try nana and determine if it's awesome or not? :p

-----Original Message-----

**Comment by andyprowl** [comment]
Cinder isn't terrible, but it's lacking in many ways AFAIR. Good for small experiments. What is this about? Are we trying a Lounge project? You guys know the chances of failure exceed the number of molecules in Puppy's mom?

**Comment by ghost** [comment]
@andyprowl I know lounge projects aren't particularly lucky, but this project is more about discussing things for the sake of it and bitching about the libs we hate, so I figured so loungers would be pretty good at it :P

**Comment by Morwenn** [comment]
Has anyone tried [Crow](https://github.com/ipkn/crow)? It is a C++ microframework for web, inspired by Flask. The presentation looks nice but I never had to actually use it, so I can't judge.

**Comment by nabijaczleweli** [comment]
On a related topic, I propose [CPR](https://github.com/whoshuu/cpr) - a C++14-isation of `libcurl`

**Comment by kbok** [comment]
@nabijaczleweli no soup for you, it's not a framework. should go to #13

**Comment by ghost** [comment]
@Morwenn crow looks nice, we should have someone try it

**Comment by nabijaczleweli** [comment]
@kbok proposed [casablanca](https://casablanca.codeplex.com) in the wrong place, moving here.
Nota bene: competes with Crow.

**Comment by nabijaczleweli** [comment]
@milleniumbug Nominated [nonius](https://github.com/rmartinho/nonius) in the wrong place, moved here.

**Comment by Morwenn** [comment]
Maybe we could create a separate topic for benchmarking frameworks. Just like testing framework, they are somewhat different from regular frameworks and libraries: it's not code that makes yoour project run, just additional code to make sure it's cool.

**Comment by nabijaczleweli** [comment]
@Morwenn Don't "maybe", do it

**Comment by milleniumbug** [comment]
Not sure we need so many topics, with the rate of actually dank stuff we add, it's unlikely that we have more than 5 items per category.

**Comment by Morwenn** [comment]
Even so, just look at that evergrowing list of messages.

**Comment by ghost** [comment]
@nabijaczleweli Agreed about casablanca: it's a framework. I disagree about it competing with crow though. casablanca it mostly about writing clients, and the only server part is the platform; crow provides a web framework, so MVC, routing, auth etc. the overlap is minimal.

**Comment by obiltschnig** [comment]
POCO C++ Libraries http://pocoproject.org. Fits decent, actually used and appreciated in industry ;-)

**Comment by no-longer-on-githu-b** [comment]
Yes POCO is great.
